src/plugins/data/public/ui/query_string_input/query_string_input.tsx

```diff
@@ -119,24 +120,34 @@ export default class QueryStringInputUI extends Component<Props, State> {
     return toUser(this.props.query.query);
   };
 
-  private fetchIndexPatterns = async () => {
+  private fetchIndexPatterns = debounce(async () => {
```

Comment:
@sulemanof could you please add some comment here why we need to add `debounce` here